src/v/utils/delta_for.h

```diff
@@ -18,6 +18,11 @@
 
 #include <seastar/util/log.hh>
 
+#include <concepts>
```

Comment:
```
- changes std::array to std::span(const) parameters when possible
- decoding now zeros the out value (inexpensive)
- pack* unpack* functions are replaces with only one leverging constexpr
  programming. a namespace details::decomp contains some helpers that
  describes how N_BITS gets subdivided in unsigned values and spare
  bits, for (de)serialization
- multiple writes/reads to the underlying iobuf are replaces with a
  single one. this enables some performance gains
```

is this 4 separate commits?

Reply:
All in one. Basically nr 3 is the driver of this commit, 1 2 and 4 are because of how i wrote pack/unpack.